TideApp/TideApp/Scenes/TideTimesView/SubViews/WaterTemperatureView.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  WaterTemperatureView.swift
+//  TideApp
+//
+//  Created by Dan Smith on 19/05/2021.
+//
+
+import SwiftUI
+
+struct WaterTemperatureView: View {
+  @State var temperature: Int
+
+  var body: some View {
+    HStack {
+      AnimatedDataLabel(title: "Water", value: temperature, symbol: "℃", systemAsset: .thermometer)
+      ThermometerBarView(temperature: temperature)
+    }
+  }
+}
+
+// MARK: - AnimatedDataLabel -
+struct AnimatedDataLabel: View {
+  @State var appearing = false
+  let title: String
+  let value: Int
+  let symbol: String
+  let systemAsset: SystemAsset?
+  let delay: Double
+
+  init(title: String, value: Int, symbol: String = "", delay: Double = 1, systemAsset: SystemAsset?) {
+    self.title = title
+    self.value = value
+    self.symbol = symbol
+    self.delay = delay
+    self.systemAsset = systemAsset
+  }
+
+  var body: some View {
+    VStack {
+      HStack {
+        Text(title)
+        .font(.title3)
+
+        if let systemAsset = systemAsset {
+          Image(.thermometer)
+        }
+      }
+
+      Text("\(value)\(symbol)")
+        .font(.largeTitle)
+        .scaleEffect(appearing ? 1 : 0)
+        .animation(Animation.spring(response: 0.2,
+                                    dampingFraction: 0.4,
+                                    blendDuration: 0.7)
+                    .delay(delay))
+    }
+    .onAppear {
+      withAnimation {
+        appearing.toggle()
+      }
+    }
+    .foregroundColor(.titleColor)
+  }
+}
+
+// MARK: - ThermometerBarView -
+struct ThermometerBarView: View {
+  @State var temperature: Int
+  @State var appearing = false
+
+  var maxTemp: Int = 26 // NB: The highest ever recorded sea temperature was 25.9 ℃
```

Comment:
// NB: The highest ever recorded sea temperature was 25.9 ℃
Yeah, but with the Global Warming???? (Joking.. well not a really funny joke 😢 😭 )

Reply:
👍 Hopefully it won't go higher, have a safeMax elsewhere in case we do hit higher than 26 🙀...